From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Nicolas Iooss <nicolas.iooss_linux@m4x.org>
Date: Sat, 9 Sep 2017 18:17:40 +0200
Subject: [PATCH] {BUG in 4.14-rc1} drivers/infiniband/hw: fix use of
 uninitialized variable

clang reports:

    drivers/infiniband/hw/hfi1/rc.c:855:6: error: variable 'bth1' is
    used uninitialized whenever 'if' condition is false
    [-Werror,-Wsometimes-uninitialized]
            if (qp->s_mig_state == IB_MIG_MIGRATED)
                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    drivers/infiniband/hw/hfi1/rc.c:857:37: note: uninitialized use
    occurs here
            hfi1_make_bth_aeth(qp, ohdr, bth0, bth1);
                                               ^~~~
    drivers/infiniband/hw/hfi1/rc.c:855:2: note: remove the 'if' if its
    condition is always true
            if (qp->s_mig_state == IB_MIG_MIGRATED)
            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    drivers/infiniband/hw/hfi1/rc.c:818:16: note: initialize the
    variable 'bth1' to silence this warning
            u32 bth0, bth1;
                          ^
                           = 0
---
 drivers/infiniband/hw/hfi1/rc.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/infiniband/hw/hfi1/rc.c b/drivers/infiniband/hw/hfi1/rc.c
index fd01a760259f..702c93108425 100644
--- a/drivers/infiniband/hw/hfi1/rc.c
+++ b/drivers/infiniband/hw/hfi1/rc.c
@@ -814,7 +814,7 @@ static inline void hfi1_make_rc_ack_16B(struct rvt_qp *qp,
 	struct hfi1_pportdata *ppd = ppd_from_ibp(ibp);
 	struct hfi1_16b_header *hdr = &opa_hdr->opah;
 	struct ib_other_headers *ohdr;
-	u32 bth0, bth1;
+	u32 bth0, bth1 = 0; /* BUG */
 	u16 len, pkey;
 	u8 becn = !!is_fecn;
 	u8 l4 = OPA_16B_L4_IB_LOCAL;
-- 
